src/query/builder.js

```diff
@@ -57,9 +57,16 @@ assign(Builder.prototype, {
     return cloned;
   },
 
-  timeout(ms) {
+  timeout(ms, {cancel} = {}) {
     if(isNumber(ms) && ms > 0) {
       this._timeout = ms;
+      if (cancel) {
```

Comment:
Unnecessary nested if-statements here. `if (cancel && this.client.canCancelQuery)`